modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/PointStackerProcess.java

```diff
@@ -267,7 +273,7 @@ private Coordinate getStackedPointLocation(PreserveLocation preserveLocation, St
                     stkPt = new StackedPoint(indexPt, new Coordinate(centreX, centreY));
                     stackedPts.put(stkPt.getKey(), stkPt);
                 }
-                stkPt.add(pout);
+                stkPt.add(pout,weightClusterPosition);
```

Comment:
Is there a case where one would want to add one point as weighted and another as nearest, into the same StackedPoint? If not, maybe better to add the flag as a field in StackedPoint, and set it via a setter. The add method interface would not change, only its implementation.